homeassistant/components/tasmota/config_flow.py

```diff
@@ -0,0 +1,52 @@
+"""Config flow for Tasmota."""
+from collections import OrderedDict
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.mqtt import valid_subscribe_topic
+
+from .const import (  # pylint:disable=unused-import
+    CONF_DISCOVERY_PREFIX,
+    DEFAULT_PREFIX,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class FlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        return await self.async_step_config()
+
+    async def async_step_config(self, user_input=None):
+        """Confirm the setup."""
+        errors = {}
+
+        if user_input is not None:
+            bad_prefix = False
+            try:
+                prefix = user_input[CONF_DISCOVERY_PREFIX]
+                valid_subscribe_topic(f"{prefix}/#")
+            except vol.Invalid:
+                errors["base"] = "invalid_discovery_topic"
+                bad_prefix = True
+            if not bad_prefix:
+                return self.async_create_entry(title="Tasmota", data=user_input)
+
+        fields = OrderedDict()
+        fields[vol.Optional(CONF_DISCOVERY_PREFIX, default=DEFAULT_PREFIX)] = str
```

Comment:
Is this a case that is common and we should support? Or should we only support this in advanced mode? 

```python
if self.show_advanced_options:
    ...
```

Reply:
Right, it makes sense to keep this as an advanced option, especially in light of integration discovery.